From 253095aa04beeabf5a80553c6e5d92a4c1ab9e2e Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Wed, 13 Jan 2016 14:50:29 +0800
Subject: [PATCH 0836/5242] MLK-12189 imx: mx7d correct iomux error for SAI3
 and ENET2_EN

commit  928dd9bb0b6b1c9d3f68786f6b9bb2cd425d53ce from
https://source.codeaurora.org/external/imx/linux-imx.git

pinmux settings using GPIO1_IO0[0-7] should use iomuxc_lpsr,
but not iomuxc. If use iomuxc, you will set wrong register
and may impact other functions.

Without this patch, SAI3_MCLK use GPIO1_IO03 pinmux and impacts
QSPI function.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-sdb.dts |   21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index bfd6d4a..d895a4b 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -600,7 +600,7 @@
 
 &sai3 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sai3>;
+	pinctrl-0 = <&pinctrl_sai3 &pinctrl_sai3_mclk>;
 	assigned-clocks = <&clks IMX7D_SAI3_ROOT_SRC>,
 			  <&clks IMX7D_SAI3_ROOT_CLK>;
 	assigned-clock-parents = <&clks IMX7D_PLL_AUDIO_POST_DIV>;
@@ -641,6 +641,18 @@
 				MX7D_PAD_LPSR_GPIO1_IO07__GPIO1_IO7	  0x14
 			>;
 		};
+
+		pinctrl_enet2_epdc0_en: enet2_epdc0_grp {
+			fsl,pins = <
+				MX7D_PAD_LPSR_GPIO1_IO04__GPIO1_IO4	0x80000000
+			>;
+		};
+
+		pinctrl_sai3_mclk: sai3grp_mclk {
+			fsl,pins = <
+				MX7D_PAD_LPSR_GPIO1_IO03__SAI3_MCLK	0x1f
+			>;
+	       };
 	};
 };
 
@@ -838,12 +850,6 @@
 			>;
 		};
 
-		pinctrl_enet2_epdc0_en: enet2_epdc0_grp {
-			fsl,pins = <
-				MX7D_PAD_LPSR_GPIO1_IO04__GPIO1_IO4          0x80000000
-			>;
-		};
-
 		pinctrl_epdc0: epdcgrp0 {
                         fsl,pins = <
 				MX7D_PAD_EPDC_DATA00__EPDC_DATA0  0x2
@@ -990,7 +996,6 @@
 
 		pinctrl_sai3: sai3grp {
 			fsl,pins = <
-				MX7D_PAD_LPSR_GPIO1_IO03__SAI3_MCLK    0x1f
 				MX7D_PAD_UART3_TX_DATA__SAI3_TX_BCLK   0x1f
 				MX7D_PAD_UART3_CTS_B__SAI3_TX_SYNC     0x1f
 				MX7D_PAD_UART3_RTS_B__SAI3_TX_DATA0    0x30
-- 
1.7.9.5

